core/src/calleeFunctions/index.ts

```diff
@@ -24,14 +36,65 @@ export const getCalleeData = async function (
     network: string,
     collateralType: string,
     marketId: string,
-    profitAddress: string
+    profitAddress: string,
+    preloadedPools?: Pool[]
 ): Promise<string> {
     const collateral = getCollateralConfigByType(collateralType);
     const marketData = collateral.exchanges[marketId];
     if (!marketData || !marketData.callee || !allCalleeFunctions[marketData.callee]) {
         throw new Error(`Unsupported collateral type "${collateralType}"`);
     }
-    return await allCalleeFunctions[marketData.callee].getCalleeData(network, collateral, marketId, profitAddress);
+    return await allCalleeFunctions[marketData.callee].getCalleeData(
+        network,
+        collateral,
+        marketId,
+        profitAddress,
+        preloadedPools
+    );
+};
+const getCalleeAutoRoute = async (
+    network: string,
+    collateral: CollateralConfig,
+    marketId: string,
+    amount: BigNumber = new BigNumber('1')
+): Promise<string[]> => {
+    const calleeConfig = collateral.exchanges[marketId];
+    if (!('automaticRouter' in calleeConfig)) {
+        throw new Error('Unexpected market data parameters. Failed to determine the callee type.');
+    }
+    return (await fetchAutoRouteInformation(network, collateral.symbol, amount.toFixed())).route || [];
+};
+
+export const getPools = async (
+    network: string,
+    collateral: CollateralConfig,
+    marketId: string,
+    amount: BigNumber = new BigNumber(1)
+) => {
+    const calleeConfig = collateral.exchanges[marketId];
+    if (calleeConfig.callee === 'UniswapV2LpTokenCalleeDai') {
+        return undefined;
+    }
+    const route =
+        'route' in calleeConfig ? calleeConfig.route : await getCalleeAutoRoute(network, collateral, marketId, amount);
+    return await routeToPool(network, route, collateral.symbol);
```

Comment:
As pointed out in the issue several times: we CAN NOT assume that the route provided from the auto router will ONLY contain symbols that are known to the maker. Therefore we CAN NOT convert automatic route to pool. Instead, the information provided by the AlphaRouter should contain enough information to build a pool object: addresses, symbols and fees.